### PR TITLE
Add characterization tests for basic langchain text responses and stream data.

### PR DESCRIPTION
**Pull request by lgrammel**
## Overview
- Introduce `x-flush-delay` header in mock server to speed up tests (default to 5ms to speed up existing tests)
- Introduce port configuration for mock server to allow running several tests in parallel
- Add langchain dev dependency
- Add langchain characterization tests for existing behavior based on examples from website

## Note
The langchain stream test does not work in the edge environment and is deactivated in that environment. The errror is caused by the following chain of events:

* langchain functionality is required for langchain integration testing, so I added a dev dependency
* langchain is used in other parts of the project, therefore the dependencies become non-dev dependencies in the lockfile
* langchain has a dependency on uuidjs/uuid
* uuidjs/uuid has an open bug with their dist: https://github.com/uuidjs/uuid/issues/678 (that’s open for almost a year)
*  the edge environment testing does not compile with

Details:
```
    /Users/lgrammel/repositories/vercel-ai/node_modules/.pnpm/uuid@9.0.0/node_modules/uuid/dist/esm-browser/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { default as v1 } from './v1.js';
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'
```

### CODE REVIEW

File: packages/core/streams/langchain-stream.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        },
+      },
+    });
+
+    llm.call([new HumanMessage('hello')], {}, [handlers]).catch(console.error);
+
+    const response = new StreamingTextResponse(stream);
+
+    expect(await readAllChunks(response)).toEqual([
+      '',
```
Comment (lgrammel) [inline packages/core/streams/langchain-stream.test.ts:48]: With Langchain, there is an empty text chunk at the beginning and the end. Is this desired?

**Comment by lgrammel** [comment]
The error is caused by the following chain of events:
* langchain functionality is required for langchain integration testing, so I added a dev dependency
* langchain is used in other parts of the project, therefore the dependencies become non-dev dependencies in the lockfile
* langchain has a dependency on uuidjs/uuid
* uuidjs/uuid has an open bug with their dist: https://github.com/uuidjs/uuid/issues/678 (that’s open for almost a year)
*  the edge environment testing does not compile with

```
    Details:

    /Users/lgrammel/repositories/vercel-ai/node_modules/.pnpm/uuid@9.0.0/node_modules/uuid/dist/esm-browser/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { default as v1 } from './v1.js';
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'
```

I’ve tried setting up Jest transformIgnorePatterns, but it does not help.

**Comment by lgrammel** [comment]
> The error is caused by the following chain of events:
> 
> * langchain functionality is required for langchain integration testing, so I added a dev dependency
> * langchain is used in other parts of the project, therefore the dependencies become non-dev dependencies in the lockfile
> * langchain has a dependency on uuidjs/uuid
> * uuidjs/uuid has an open bug with their dist: [[BUG] `Unexpected token 'export'` with Jest (on Cloudflare Workers) uuidjs/uuid#678](https://github.com/uuidjs/uuid/issues/678) (that’s open for almost a year)
> * the edge environment testing does not compile with
> 
> ```
>     Details:
> 
>     /Users/lgrammel/repositories/vercel-ai/node_modules/.pnpm/uuid@9.0.0/node_modules/uuid/dist/esm-browser/index.js:1
>     ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { default as v1 } from './v1.js';
>                                                                                       ^^^^^^
> 
>     SyntaxError: Unexpected token 'export'
> ```
> 
> I’ve tried setting up Jest transformIgnorePatterns, but it does not help.

Addressed by changing the test file to `.cts`

**Comment by lgrammel** [comment]
Tests are not run currently.

File: packages/core/streams/langchain-stream.test.ts (hunk 1 above)
Comment (MaxLeiter) [reply packages/core/streams/langchain-stream.test.ts:54]: No, but it shouldn't be a problem and not worth investigating until we finalize the complex mode API imo

**Review by MaxLeiter — APPROVED** [review]

**Pull request merged**